lib/Graph/Graph.cpp

```diff
@@ -1604,6 +1604,28 @@ SaveNode *Function::createSave(Context &ctx, llvm::StringRef name,
   return addNode(new SaveNode(name, input, dest));
 }
 
+Node *Function::createDotProduct(llvm::StringRef name, NodeValue X,
+                                 NodeValue Y) {
+  auto XDimsSize = X.dims().size();
+  auto YDimsSize = Y.dims().size();
+
+  // X and Y must have the same shape.
```

Comment:
This comment needs to be inside assert error message:
assert(XDimsSize == YDimsSize && "X and Y must have the same shape.");